src/utils.ts

```diff
@@ -207,6 +207,20 @@ export function getDayDifference(date1: Date, date2: Date): number {
   return Math.ceil(diffTime / (1000 * 60 * 60 * 24));
 }
 
+export function getDateFromString(dateString: string): Date {
+  if (dateString.indexOf("/") > -1) {
+    // probably this format: 8/25/21: m/d/y
```

Comment:
I tried that before. But it is strange. 

New Date from a m/d/y returns this date:
```js
const dateString = "8/25/21";
console.log(new Date(dateString));   //returns 2021-08-24T22:00:00.000Z
```

But new Date from a yyyy-mm-dd returns this:
```js
const dateString = "2021-08-25";
console.log(new Date(dateString));   //returns 2021-08-25T00:00:00.000Z
```

This led to inconsistent date entries in the history result array. Some entries have been at 22pm. The others not. And that's why I tried to bring the dates into the same source format.

Reply:
This little js snippet shows the problem:
![Bildschirmfoto 2022-02-09 um 20 30 28](https://user-images.githubusercontent.com/956016/153276132-5ebe7ef3-58aa-42ae-949e-0b7eef6341fc.png)

Reply:
Okay you are right! that is because your computer is running in timerzone GMT +2.
The Dockercontainer is running in GMT and then both dates are the same.
there are serval points the system is reacting in this way!
And i agree that your way is the way with more consistency, so let it as it is. 
But the function is only needed in frozen-incidence.ts so in my opinion it makes more sence to put that funtion in frozen-incidence.ts and not in utils.ts!
Und unfortnatly i found another little bug
to decide if the archive data must be loaded you compare the length of the history with the number of days that are given as the request parameter days. (as one part of the decisiontree)
`districts[0].history.length < days`
per definition the days parameter is from today x days in the past.
Unfortunately the RKI is not providing the frozen-incidence value every day, they update the excelsheet only mondays.
The count of historyentry is different to that day parameter!
the comparment should be
compare the date of the lowest date in the history with the date x days in the past from today!
`districts[0].history[0].date < new Date(getDateBefore(days))`
for both states and districts

Reply:
Ok. I will change that. But I think that it should be the other way:
`districts[0].history[0].date > new Date(getDateBefore(days))`

Reply:
tell me why do you think so?

Reply:
Well. If the first (so the minimum) date in the current history is array is newer than our desired target date, then we should fetch the archive.

Reply:
of corse you are rigth. I just debug this and i wonder why the archiv data will be loaded wenn the parameter is 14 days!
there is another bug since a long time!
change also the first days filter 
```
      history = history.filter((element) > element.date > reference_date);
```
in row 241 this is the old bug and 272 also 
to 
```
      history = history.filter((element) => element.date >= reference_date);

```
and for districts also
row 104 and 131
please

Reply:
But this will give 4 history entries when days is 3. Or is this is also some problem with not running in a Docker Container?
![image](https://user-images.githubusercontent.com/956016/153290044-fb3d09a5-7396-471e-9f79-518ae9850b2f.png)

Reply:
pfffff i have only 2 entrys
![2022-02-09 21_12_20-Mozilla Firefox](https://user-images.githubusercontent.com/6493772/153291317-d7683fe8-4d2b-4114-9eb8-526d7b51a4e2.png)
dammed :-)
Today is the 9th 3 days ago is the 6th so I would actually expect the 6th, 7th and 8th and, if available, the 9th. but the 9th is not in the file. But why am I missing the 6th??

Reply:
sorry forget about it
i have 3 entrys
![2022-02-09 21_24_30-Mozilla Firefox](https://user-images.githubusercontent.com/6493772/153292587-9dd66e14-4390-4ab1-8dde-96a8a7e15e51.png)

Reply:
and that is what i expect

Reply:
change your timezone to GMT i i would expect that you have the same result

Reply:
sorry you cant see the whole address
![2022-02-09 21_27_51-Mozilla Firefox](https://user-images.githubusercontent.com/6493772/153293022-c42be99e-7b96-4b71-a6a9-85888ba7211a.png)

Reply:
So. I am a bit confused now. Should I change the `>` to `>=`? Or keep it the way it is?

Reply:
i am confused too :-)
i think to not change the logic witch dates are shown DONT change `>` to `>=`
but to not alwasy load the archiv data change 
`states[0].history[0].date > new Date(getDateBefore(days));`
to
`states[0].history[0].date > new Date(getDateBefore(days - 1));`

districts also

Reply:
Done

Reply:
Thanks very much